src/workspace/discovery.rs

```diff
@@ -0,0 +1,221 @@
+use std::{path::PathBuf, sync::Arc};
+
+use itertools::Itertools;
+use miette::{Diagnostic, NamedSource, Report};
+use pixi_consts::consts;
+use pixi_manifest::{
+    utils::WithSourceCode, ExplicitManifestError, LoadManifestsError, Manifests, TomlError,
+    WarningWithSource, WithWarnings, WorkspaceDiscoveryError,
+};
+use thiserror::Error;
+
+use crate::workspace::Workspace;
+
+/// Defines where the search for the workspace should start.
+#[derive(Debug, Clone, Default)]
+pub enum DiscoveryStart {
+    /// Start the search from the current directory indicated by
+    /// [`std::env::current_dir`].
+    #[default]
+    CurrentDir,
+
+    /// Start the search from the given directory.
+    SearchRoot(PathBuf),
+
+    /// Use the manifest file at the given path. Only search for a workspace if
+    /// the specified manifest is not a workspace.
+    ExplicitManifest(PathBuf),
```

Comment:
Do we need both?

If yes, at least the doc string of `ExplicitManifest` should be updated since the path can also be a directory containing a manifest.

Reply:
Yes `ExplicitManifest` errors if the specified path doesnt refer to a manifest. I added some documentation to highlight that detail.